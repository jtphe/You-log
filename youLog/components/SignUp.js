import React from "react";
import {
  StyleSheet,
  Text,
  TextInput,
  View,
  Keyboard,
  ScrollView
} from "react-native";
import { Button } from "react-native-paper";
import Toast from "react-native-root-toast";
// import * as SQLite from "expo-sqlite";

/**
 * The SignUp class
 */
class SignUp extends React.Component {
  /**
   * Pass the title option to the navigation
   */
  static navigationOptions = {
    title: "Inscription"
  };

  /**
   * name: The user name | 
   * email: The user email |
   * password: The user password
   * @type {{name: string, email: string, password: string}}
   */
  state = {
    name: "",
    email: "",
    password: ""
  };

  /**
   * Handle the change of name
   *  @param {string} text - The name to change
   */
  _onChangeName = text => {
    this.setState({
      name: text
    });
  };

  /**
   * Handle the change of email
   *  @param {string} text - The email to change
   */
  _onChangeEmail = text => {
    this.setState({
      email: text
    });
  };

  /**
   * Handle the change of password
   *  @param {string} text - The password to change
   */
  _onChangePassword = text => {
    this.setState({
      password: text
    });
  };

  /**
   * Insert the user in the SQLite DB
   * @param  {string} name - The name of the user
   * @param  {string} email - The email of the user
   * @param  {string} password - The password of the user
   */
  _insert = (name, email, password) => {
    const usersDB = SQLite.openDatabase("users.db");
    usersDB.transaction(tx => {
      tx.executeSql(
        "CREATE TABLE IF NOT EXISTS users (id integer primary key not null, name text, email text, password text);"
      );
    });

    const query = "INSERT INTO users(name, email, password) VALUES (?,?,?);";
    const usr = [name, email, password];

    usersDB.transaction(tx => {
      tx.executeSql(query, usr);
    });
  };

  /**
   * Check the value in the text input
   * @param  {string} email - Email of the user
   * @param  {string} password - Password of the user
   * @param  {string} name - Name of the user
   */
  _checkTextInput = (email, password, name) => {
    Keyboard.dismiss();
    let reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;

    if (
      reg.test(email) === false &&
      !email.length > 0 &&
      !password.length > 0
    ) {
      Toast.show("Entrez un email valide", {
        duration: Toast.durations.LONG,
        position: Toast.positions.TOP + 60,
        shadow: false,
        opacity: 1
      });
    } else {
      this._insert(name, email, password);
      this.props.navigation.navigate("SignIn");
    }
  };

  /**
   * Render the sign up screen
   * @returns {React.Component} - SignUp Component
   */
  render() {
    const { name, email, password } = this.state;
    return (
      <View style={styles.container}>
        <Text style={styles.signUpTitle}>Cr√©ez votre compte</Text>
        <View style={styles.textInputContainer}>
          <ScrollView>
            <TextInput
              style={styles.input}
              placeholder="Nom"
              onChangeText={this._onChangeName}
              autoCapitalize="none"
              returnKeyType={"next"}
              onSubmitEditing={() => {
                this.emailTextInput.focus();
              }}
              blurOnSubmit={false}
            />
            <TextInput
              ref={input => {
                this.emailTextInput = input;
              }}
              style={styles.input}
              placeholder="E-mail"
              textContentType="emailAddress"
              keyboardType="email-address"
              onChangeText={this._onChangeEmail}
              autoCapitalize="none"
              returnKeyType={"next"}
              onSubmitEditing={() => {
                this.passwordTextInput.focus();
              }}
              blurOnSubmit={false}
            />
            <TextInput
              ref={input => {
                this.passwordTextInput = input;
              }}
              style={styles.input}
              placeholder="Mot de passe"
              textContentType="password"
              secureTextEntry={true}
              onChangeText={this._onChangePassword}
            />
          </ScrollView>
        </View>
        <Button
          mode={"contained"}
          labelStyle={{ color: "#fff" }}
          style={styles.btnSignIn}
          onPress={() =>
            this._checkTextInput(email.trim(), password, name.trim())
          }
        >
          Valider
        </Button>
      </View>
    );
  }
}

/**
 * Styles of the SignUp Component
 */
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    alignItems: "stretch",
    justifyContent: "center"
  },
  textInputContainer: {
    marginTop: 20
  },
  input: {
    height: 50,
    marginLeft: 50,
    marginRight: 50,
    paddingLeft: 15,
    borderColor: "#ecf0f1",
    borderWidth: 1,
    borderRadius: 6,
    marginBottom: 25
  },
  btnSignIn: {
    padding: 8,
    paddingLeft: 35,
    paddingRight: 35,
    marginLeft: 50,
    marginRight: 50
  },
  signUpTitle: {
    textAlign: "center",
    fontSize: 20,
    marginBottom: 10
  }
});

export default SignUp;
