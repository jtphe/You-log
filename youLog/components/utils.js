global.fetch = require('jest-fetch-mock');
import { Keyboard } from 'react-native';
/**
 * Insert the user in the SQLite DB
 * @param  {string} name - The name of the user
 * @param  {string} email - The email of the user
 * @param  {string} password - The password of the user
 */
export const _insert = async (name, email, password) => {
    try {
        var form = new FormData();
        form.append("name", name);
        form.append("mail", email);
        form.append("password", password);
        await fetch(
            "http://jdevalik.fr/api/insertuser.php",
            {
                method: "POST",
                header: {
                    Accept: "application/json",
                    "Content-Type": "'multipart/form-data"
                },
                body: form
            }
        );
    } catch (error) {
        console.log("Error while inserting user", error);
    }
};

/**
 * Select all the users from the API
 */
export const _loadUsers = async () => {
    try {
        let response = await fetch(
            "http://jdevalik.fr/api/getusers.php",
            {
                method: "GET",
                header: {
                    Accept: "application/json",
                    "Content-Type": "application.json"
                }
            }
        );
        let responseJs = await response.json();
        this.setState({
            users: responseJs
        });
    } catch (error) {
        console.log("Error while loading user", error);
    }
};

/**
* Update the password in the SQLite DB
* @param {string} text - The new password
*/
export const _updatePassword = async () => {
    try {
        var form = new FormData();
        form.append("mail", this.state.email);
        form.append("password", this.state.password);
        await fetch("http://jdevalik.fr/api/updateuser.php", {
            method: "POST",
            header: {
                Accept: "application/json",
                "Content-Type": "'multipart/form-data"
            },
            body: form
        });
    } catch (error) {
        console.log("Error while inserting user", error);
    }
    this.setState({
        visible: false,
        canUpdatePassword: false
    });
    Keyboard.dismiss();
};