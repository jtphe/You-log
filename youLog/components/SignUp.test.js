import React from 'react';
import SignUp from './SignUp';
import renderer from 'react-test-renderer';
import { _insert } from './utils';

describe('SIGN UP TESTS', () => {
    test('[SNAPSHOT]: SIGN UP TEST', () => {
        const tree = renderer.create(<SignUp />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    beforeEach(() => {
        window.fetch = jest.fn().mockRejectedValue({});
    });

    test('[MOCKUP]: INSERT USER TEST', async () => {
        await _insert("Jean", "jean@phe.fr", "testtest")
        expect(window.fetch).toHaveBeenCalledWith("http://jdevalik.fr/api/insertuser.php");
    })
})