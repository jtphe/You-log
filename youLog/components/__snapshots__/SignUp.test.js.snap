// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIGN UP TESTS [SNAPSHOT]: SIGN UP TEST 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Créez votre compte
  </Text>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <RCTScrollView>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          blurOnSubmit={false}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Nom"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "borderColor": "#ecf0f1",
              "borderRadius": 6,
              "borderWidth": 1,
              "height": 50,
              "marginBottom": 25,
              "marginLeft": 50,
              "marginRight": 50,
              "paddingLeft": 15,
            }
          }
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          blurOnSubmit={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="E-mail"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "borderColor": "#ecf0f1",
              "borderRadius": 6,
              "borderWidth": 1,
              "height": 50,
              "marginBottom": 25,
              "marginLeft": 50,
              "marginRight": 50,
              "paddingLeft": 15,
            }
          }
          textContentType="emailAddress"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mot de passe"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#ecf0f1",
              "borderRadius": 6,
              "borderWidth": 1,
              "height": 50,
              "marginBottom": 25,
              "marginLeft": 50,
              "marginRight": 50,
              "paddingLeft": 15,
            }
          }
          textContentType="password"
          underlineColorAndroid="transparent"
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "marginLeft": 50,
        "marginRight": 50,
        "minWidth": 64,
        "padding": 8,
        "paddingLeft": 35,
        "paddingRight": 35,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "color": "#fff",
                },
              ],
            ]
          }
        >
          Valider
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`SIGN UP TESTS Logged test 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Créez votre compte
  </Text>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <RCTScrollView>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          blurOnSubmit={false}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Nom"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "borderColor": "#ecf0f1",
              "borderRadius": 6,
              "borderWidth": 1,
              "height": 50,
              "marginBottom": 25,
              "marginLeft": 50,
              "marginRight": 50,
              "paddingLeft": 15,
            }
          }
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          blurOnSubmit={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="E-mail"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "borderColor": "#ecf0f1",
              "borderRadius": 6,
              "borderWidth": 1,
              "height": 50,
              "marginBottom": 25,
              "marginLeft": 50,
              "marginRight": 50,
              "paddingLeft": 15,
            }
          }
          textContentType="emailAddress"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mot de passe"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#ecf0f1",
              "borderRadius": 6,
              "borderWidth": 1,
              "height": 50,
              "marginBottom": 25,
              "marginLeft": 50,
              "marginRight": 50,
              "paddingLeft": 15,
            }
          }
          textContentType="password"
          underlineColorAndroid="transparent"
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "marginLeft": 50,
        "marginRight": 50,
        "minWidth": 64,
        "padding": 8,
        "paddingLeft": 35,
        "paddingRight": 35,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "color": "#fff",
                },
              ],
            ]
          }
        >
          Valider
        </Text>
      </View>
    </View>
  </View>
</View>
`;
